mcp-server/main.py

```
@@ -360,36 +226,148 @@
+            )
+            
+            logger.info(f"Legacy REST endpoint returning {len(context_items)} context items")
+            return response
+                
+        except Exception as e:
+            logger.error(f"Error processing context request: {str(e)}", exc_info=True)
+            raise
+        
+    except ValueError as e:
+        logger.error(f"Validation error in legacy endpoint: {str(e)}")
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)